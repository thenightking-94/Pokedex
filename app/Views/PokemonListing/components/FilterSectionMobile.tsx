import { Backdrop, Grid, Modal } from '@mui/material';
import { ReactElement, useState } from 'react';
import CloseIcon from '../../Icons-svg/closeIcon';
import FilterIcon from '../../Icons-svg/filterIcon';
import ExpandIcon from '../../Icons-svg/expandIcon';
import { formatPokemonText } from '../../../Utils/Utils';
  

export default function FilterSectionMobile (props): ReactElement{
    const {handleChangeGenderMobile, handleChangeTypeMobile, PokemonTypes, PokemonGender, pokemonGenderDataFiltered, pokemonTypeDataFiltered} = props
    const [toggleModal, setToggleModal] = useState(false)
    const [toggleTypeFilters, setToggleTypeFilters] = useState(false)
    const [toggleGenderFilters, setToggleGenderFilters] = useState(false)
   
    const toggleFilterModal = () => {
        !!toggleModal ? setToggleModal(false) : setToggleModal(true)
    }
    const openTypeFilters = () => {
        !!toggleTypeFilters ? setToggleTypeFilters(false) : setToggleTypeFilters(true)
    }
    const openGenderFilters = () => {
        !!toggleGenderFilters ? setToggleGenderFilters(false) : setToggleGenderFilters(true)
    }

    return(
        <div className='filter-section'>
              <div onClick={toggleFilterModal} className='filterIcon'>
                  <FilterIcon />
             </div>
            {
                !!toggleModal &&
                    <Modal
                        aria-labelledby="transition-modal-title"
                        aria-describedby="transition-modal-description"
                        open={toggleModal}
                        onClose={toggleFilterModal}
                        closeAfterTransition
                        BackdropComponent={Backdrop}
                        BackdropProps={{
                        timeout: 500,
                        }}
                    >
                            <div className='filter-modal'>
                               <div className='heading'>
                                    <p className='label'>Filters</p>
                                    <div onClick={toggleFilterModal} className='close-icon'>
                                        <CloseIcon/>
                                    </div>
                               </div>

                               <div className='type-filter'>
                                    <Grid className='type-grid' container spacing={2} alignItems='center' justifyContent={'space-between'}> 
                                        <Grid item sm={3} className='label'>Type</Grid>
                                        <Grid item sm={6} className='labels' ><p className='label-type'>{'(Normal'}</p><p className='bold-type'>{' +5 More)'}</p></Grid>
                                        <Grid onClick={openTypeFilters} item sm={3} className='expand-icon'>
                                            {toggleTypeFilters? <CloseIcon/> : <ExpandIcon/>}
                                        </Grid>
                                    </Grid>
                                    {!!toggleTypeFilters && 
                                        <div className='checkbox-block'>
                                            <Grid container spacing={2}>
                                                {PokemonTypes?.map((i, index)=> {
                                                    return(
                                                        <Grid className='checkbox-el' key={index} item xs={6}>
                                                            {
                                                                !pokemonTypeDataFiltered.includes(i)?
                                                                <input id={i} onChange={(e)=>{handleChangeTypeMobile(e.target.id)}} type='checkbox' value={i} name={i}/>
                                                                    :
                                                                <input checked id={i} onChange={(e)=>{handleChangeTypeMobile(e.target.id)}} type='checkbox' value={i} name={i}/>
                                                            }
                                                            <label className='label-input' htmlFor={i}>{formatPokemonText(i)}</label>
                                                        </Grid>
                                                    )
                                                })}
                                            </Grid>
                                        </div>
                                    }
                               </div>
                               <div className='type-filter'>
                                    <Grid className='gender-grid' container spacing={2} alignItems='center' justifyContent={'space-between'}>   
                                        <Grid item sm={3} className='label'>Gender</Grid>
                                        <Grid item sm={6} className='labels'><p className='label-gender'>{'(Male'}</p><p className='bold-gender'>{' +2 More)'}</p></Grid>
                                        <Grid onClick={openGenderFilters} item sm={3} className='expand-icon'>
                                            {toggleGenderFilters? <CloseIcon/> : <ExpandIcon/>}
                                        </Grid>
                                    </Grid>
                                    {!!toggleGenderFilters && 
                                        <div className='checkbox-block'>
                                            <Grid container spacing={2}>
                                                {PokemonGender?.map((i, index)=> {
                                                    return (<Grid className='checkbox-el' key={index} item xs={6}>
                                                                {
                                                                    !pokemonGenderDataFiltered.includes(i) ?
                                                                    <input id={i} type='checkbox' onChange={(e)=>{handleChangeGenderMobile(e.target.id)}} value={i} name={i}/>
                                                                            : 
                                                                    <input id={i} type='checkbox' checked onChange={(e)=>{handleChangeGenderMobile(e.target.id)}} value={i} name={i}/>
                                                                }
                                                                <label className='label-input' htmlFor={i}>{formatPokemonText(i)}</label>
                                                            </Grid>)
                                                })}
                                            </Grid>
                                        </div>
                                    }
                               </div>
                               <div className='type-filter'>
                                    <Grid className='stats-grid' container spacing={2} alignItems='center' justifyContent={'space-between'}>   
                                        <Grid item sm={3} className='label'>Stats</Grid>
                                        <Grid item sm={6} className='labels'><p className='label-stats'>{'(Normal'}</p><p className='bold-stats'>{' +2 More)'}</p></Grid>
                                        <Grid item sm={3} className='expand-icon'>
                                        <ExpandIcon/>
                                        </Grid>
                                    </Grid>
                               </div>
                            </div>
                       
                    </Modal>
            }
        </div>
    )

}