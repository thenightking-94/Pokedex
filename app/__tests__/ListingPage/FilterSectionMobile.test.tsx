import { render , waitFor, screen,} from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import FilterSectionMobile from '../../Views/PokemonListing/components/FilterSectionMobile'

const queryClient = new QueryClient()

describe('FilterSectionMobile', () => {
    it('renders correctly', () => {
      render(
      <QueryClientProvider client={queryClient}>
           <FilterSectionMobile/>
      </QueryClientProvider>
     )
    })
  })
describe("Jest Snapshot testing suite", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create( 
      <QueryClientProvider client={queryClient}>
        <FilterSectionMobile/>
     </QueryClientProvider>).toJSON();
      expect(domTree).toMatchSnapshot();
    });
  });