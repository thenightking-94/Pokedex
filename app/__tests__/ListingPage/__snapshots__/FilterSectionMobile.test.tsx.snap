// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="filter-section"
>
  <div
    className="filterIcon"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="58"
      viewBox="0 0 77 58"
      width="77"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="#2E3156"
        height="57.5"
        rx="8"
        width="76.6667"
      />
      <line
        stroke="white"
        x1="31.7415"
        x2="49.9955"
        y1="18.616"
        y2="18.616"
      />
      <circle
        cx="28.1922"
        cy="18.6089"
        r="3.04938"
        stroke="white"
      />
      <line
        stroke="white"
        x1="26.6709"
        x2="44.9249"
        y1="28.757"
        y2="28.757"
      />
      <circle
        cx="48.4746"
        cy="28.75"
        r="3.04938"
        stroke="white"
      />
      <line
        stroke="white"
        x1="31.7415"
        x2="49.9955"
        y1="38.8981"
        y2="38.8981"
      />
      <circle
        cx="28.1922"
        cy="38.8911"
        r="3.04938"
        stroke="white"
      />
    </svg>
  </div>
</div>
`;
